tests/unit/s2n_protocol_preferences_test.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "tls/s2n_connection.h"
+
+#include <s2n.h>
+
+#define LEN_PREFIX 1
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* input values */
+    const uint8_t protocol1[] = "protocol1";
+    const size_t protocol1_len = strlen((const char *)protocol1);
```

Comment:
Why use strlen instead of sizeof?

Reply:
`strlen` doesn't include the null byte. I guess I could do `sizeof(protocol1) - 1` but that seemed less obvious.